operators/multiclusterobservability/controllers/placementrule/placementrule_controller.go

```diff
@@ -653,7 +653,11 @@ func (r *PlacementRuleReconciler) SetupWithManager(mgr ctrl.Manager) error {
 				e.ObjectNew.GetLabels()[ownerLabelKey] == ownerLabelValue &&
 				e.ObjectNew.GetNamespace() != localClusterName &&
 				!reflect.DeepEqual(e.ObjectNew.(*mcov1beta1.ObservabilityAddon).Status.Conditions,
-					e.ObjectOld.(*mcov1beta1.ObservabilityAddon).Status.Conditions) {
+					e.ObjectOld.(*mcov1beta1.ObservabilityAddon).Status.Conditions) &&
+				!reflect.DeepEqual(e.ObjectNew.(*mcov1beta1.ObservabilityAddon).Spec,
```

Comment:
Should the whole block comparing the object spec, annotations and conditions encapsulated in a AND and contains OR?
... && (!equalSpec || !equalAnnotations || ...)

Or I don't understand something 🤷

Reply:
Nice catch